a4_test_APIweatherApp.py

```diff
@@ -0,0 +1,27 @@
+'''
+This function tests the weatherApp's class functionality
+'''
+
+import unittest
+from a4_APIweatherApp import Service
```

Comment:
You can combine these in one statement. Also you just need to import only the objects you are explicitly using. 
```
from a4_APIweatherApp import ServiceFactory
```